helm-app/templates/users-service-db.yaml

```diff
@@ -0,0 +1,43 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ .Values.usersServiceDb.statefulsetName }}
+  namespace: {{ .Values.namespace }}
+spec:
+  serviceName: "{{ .Values.usersServiceDb.service.name }}"
+  replicas: {{ .Values.usersServiceDb.replicaCount }}
+  selector:
+    matchLabels:
+      app: {{ .Values.usersServiceDb.selector.app }}
+  template:
+    metadata:
+      labels:
+        app: {{ .Values.usersServiceDb.selector.app }}
+    spec:
+      containers:
+      - name: mysql
+        image: {{ .Values.usersServiceDb.image.repository }}:{{ .Values.usersServiceDb.image.tag }}
+        ports:
+        - containerPort: {{ .Values.usersServiceDb.containerPort }}
+        env:
+        - name: MYSQL_ROOT_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: {{ .Values.secret.name }}
+              key: MYSQL_ROOT_PASSWORD
+        - name: MYSQL_DATABASE
+          valueFrom:
+            secretKeyRef:
+              name: {{ .Values.secret.name }}
+              key: MYSQL_DATABASE

```

Comment:
Because this key value has been referred from secret file, in that secret file I have provided the option to parameterized the value for key, so we can change it dynamically.